src/AstUtils.h

```diff
@@ -72,6 +72,24 @@ std::vector<const AstRecordInit*> getRecords(const AstNode& root);
  */
 std::vector<const AstRecordInit*> getRecords(const AstNode* root);
 
+/**
+ * Returns literals of a particular type in the body of a clause.
+ *
+ * @param the clause
+ * @return vector of body literals of the specified type
+ */
+// TODO (azreika): add caching
+template <typename T>
+std::vector<T*> getBodyLiterals(const AstClause& clause) {
```

Comment:
I'm not so keen on this templated method, but I don't see a better way - four almost identical methods (at least the implementation doesn't need to be in the header file) or a templated bodyLiteralVector.